Detectors/TRD/base/include/TRDBase/Tracklet.h

```diff
@@ -40,48 +40,56 @@ namespace o2
+//I assume the more info the better hence dropping the bitshift before storing in a float, but a 32bit bitshift is massive.
+//a float will kind of auto bitshift the lower bits off to construct the float.
 
 class Tracklet
 {
 
   //-----------------------------------
-  //
-  // TRD tracklet word (as from FEE)
-  // only 32-bit of information + detector ID
-  //
+  // This is essentially a debug class.
+  // It still returns the old TrackletWord of run2, rebuilt at call time.
+  // It however stores old bit limited values as floats for further downstream investigations.
   //----------------------------------
  public:
-  Tracklet(unsigned int trackletWord = 0);
-  Tracklet(unsigned int trackletWword, int hcid);
-  Tracklet(unsigned int trackletWword, int hcid, int rob, int mcm);
+  Tracklet();
+  Tracklet(int hcid);
+  Tracklet(int hcid, int rob, int mcm, float pid, float slope, float offset, float rawslope4trackletword, float rawoffset4trackletword);
   Tracklet(const Tracklet& rhs);
   ~Tracklet() = default;
 
   Tracklet& operator=(const Tracklet& o) { return *this; }
 
   // int getHCId() const { return 2 * getDetector() + (getYbin() > 0 ? 1 : 0); }
 
-  float getdZdX() const { return 0; }
+  float getdZdX() const
+  {
+    LOG(fatal) << "Tracklet::getdZdX not implemented yet ";
+    return 0.0;
+  }
 
-  void localToGlobal(float&, float&, float&, float&) {}
+  void localToGlobal(float&, float&, float&, float&) { LOG(fatal) << "Tracklet::localToGlobal not implemented yet "; }
 
   void print(std::string* /*option=""*/) const {}
 
   // ----- Getters for contents of tracklet word -----
   int getYbin() const;                                          // in units of 160 um
   int getdY() const;                                            // in units of 140 um
-  int getZbin() const { return ((mTrackletWord >> 20) & 0xf); } // in pad length units
-  int getPID() const { return ((mTrackletWord >> 24) & 0xff); }
+  int getZbin() const { return ((mROB >> 1) << 2) | (mMCM >> 2); }
 
-  // ----- Getters for MCM-tracklet information -----
+  int getPID() const { return mPID; }
+
+  // ----- Getters for tracklet information -----
   int getMCM() const { return mMCM; }
   int getROB() const { return mROB; }
-  int getLabel() const { return mLabel[0]; }
-  int getLabel(const int i) const { return mLabel[i]; }
-  const std::array<int, 3>& getLabels() const { return mLabel; }
-  bool hasLabel(const int label) const { return (mLabel[0] == label || mLabel[1] == label || mLabel[2] == label); }
+  // labels out for now, index in digits taken out so cant reconcile.
+  //  int getLabel() const { return mLabel[0]; }
+  //  int getLabel(const int i) const { return mLabel[i]; }
+  //  const std::array<int, 3>& getLabels() const { return mLabel; }
+  //  bool hasLabel(const int label) const { return (mLabel[0] == label || mLabel[1] == label || mLabel[2] == label); }
```

Comment:
is this temporary? Why take out those functions in contrast to not using them?

Reply:
labels moved to an external MCTruthContainer, although it might seem more logical to keep them with their respective tracklets.